### PR TITLE
Add threading for multiple nuimos handling

### PR DESCRIPTION
**Pull request by commento**

### CODE REVIEW

File: senic_hub/nuimo_app/__init__.py [hunk 1]
```diff
@@ -80,7 +80,11 @@ def start(self):
         self.set_active_component()
         logger.info("Connecting to Nuimo controller %s", self.controller.mac_address)
         self.controller.connect()
-        self.manager.run()
+        try:
+            self.manager.run()
+        except KeyboardInterrupt:
+            logger.debug("Stopping... Nuimo controller %s", self.controller.mac_address)
+            self.stop()
```
Comment (rudolfs) [inline senic_hub/nuimo_app/__init__.py]: Is the exception handling for handling a user pressing CTRL-C, or something like that? Would that ever happen if it's run as a thread of a daemon? Or am I off the track here?

Comment (commento) [inline senic_hub/nuimo_app/__init__.py]: this interrupt takes place also when the nuimo_app is stopped with supervisorctl

File: senic_hub/nuimo_app/__main__.py [hunk 2]
```diff
@@ -35,31 +35,30 @@ def main(config):
         config = yaml.load(f)
 
     nuimos = config['nuimos']
-    nuimo_controller_mac_address = list(nuimos.keys())[0]
 
-    if not nuimo_controller_mac_address:
+    nuimo_controllers_mac_address = list(nuimos.keys())
```
Comment (commento) [inline senic_hub/nuimo_app/__main__.py]: ok!

File: senic_hub/nuimo_app/__init__.py (hunk 1 above)
Comment (rudolfs) [inline senic_hub/nuimo_app/__init__.py]: Gotcha!

Comment (rudolfs) [inline senic_hub/nuimo_app/__init__.py]: We probably should use consistent logging levels though. The previous logger goes into info. This one should also probably go on the same level.

File: senic_hub/nuimo_app/__main__.py [hunk 3]
```diff
@@ -35,31 +35,30 @@ def main(config):
...
+
     if platform.system() == 'Linux':
         watch_config_thread = Thread(
             target=watch_config_changes, args=(config_file_path, nuimo_apps), daemon=True)
         watch_config_thread.start()
 
-    try:
-        nuimo_app.start()
-    except KeyboardInterrupt:
-        logger.debug("Stopping...")
```
Comment (commento) [inline senic_hub/nuimo_app/__main__.py]: @rudolfs I took the logging level from here

Comment (rudolfs) [inline senic_hub/nuimo_app/__main__.py]: I know, but it makes sense to have it consistent. In the other file we use info, I guess you can choose which one you prefer, but they should be the same, no?

Comment (commento) [inline senic_hub/nuimo_app/__main__.py]: it is the same exception and the same trace moved from here to `__init__.py` that's why I used debug also there. Anyway I can change it to info

File: senic_hub/nuimo_app/__main__.py [hunk 4]
```diff
@@ -35,31 +35,30 @@ def main(config):
...
     ble_adapter_name = app.registry.settings['bluetooth_adapter_name']
-    components = nuimos[nuimo_controller_mac_address]['components']
-    component_instances = get_component_instances(components)
-    nuimo_app = NuimoApp(ha_api_url, ble_adapter_name, nuimo_controller_mac_address, component_instances)
-    # TODO: create multiple NuimoApps for all Nuimos mentioned in the config file
 
     nuimo_apps = {}
-    nuimo_apps[nuimo_controller_mac_address] = nuimo_app
+
+    for nuimo_controller_mac_address in nuimo_controllers_mac_address:
```
Comment (commento) [inline senic_hub/nuimo_app/__main__.py]: 2c085d94333d27b7ea86931e88150f8b4bd28014

File: senic_hub/nuimo_app/__main__.py (hunk 2 above)
Comment (commento) [inline senic_hub/nuimo_app/__main__.py]: 2c085d94333d27b7ea86931e88150f8b4bd28014

File: senic_hub/nuimo_app/__main__.py [hunk 5]
```diff
@@ -107,6 +106,7 @@ def reload_config_file(config_path, nuimo_apps):
     for mac_addr in nuimo_apps.keys():
         if mac_addr not in nuimos:
             logger.info("A Nuimo was removed from the config file: %s", mac_addr)
+            nuimo_apps[mac_addr].stop()
             # TODO: stop NuimoApp for deleted Nuimo here
```
Comment (commento) [inline senic_hub/nuimo_app/__main__.py]: 28a1f23113f552102fb4ef5e0365c9d090aa208a

File: senic_hub/nuimo_app/__init__.py (hunk 1 above)
Comment (commento) [inline senic_hub/nuimo_app/__init__.py]: 8aab544baa3bfd26142b194d91be7aee1464cb74

**Comment by commento** [comment]
The code has been tested and does not introduce regression errors. The code presents the issue #244 still under investigation but does not introduce errors on the normal functioning. 
It has been decided to merge this PR even if this issue is present so that, in the meantime, it is possible to adapt the setup_app for multiple nuimos and test it on the master branch.

**Pull request merged**